File: Assets/DrawLine.js
#pragma strict

var target : GameObject;
private var line : LineRenderer;

function Start(){
	line = this.GetComponent.<LineRenderer>();
}

function Update () {
	var p1 : Vector3 = gameObject.transform.position;
    var p2 : Vector3 = target.transform.position;
        
    p1.z = 1;
    p2.z = 1;
        
    line.SetPosition(0, p1);
    line.SetPosition(1, p2);

	var distance : float = Vector2.Distance(transform.position, target.transform.position);
		
	line.material.mainTextureScale = new Vector2(distance, 1);
	line.sortingOrder = 10;

}

function FixedUpdate () {
}

File: Assets/DrawLine.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class DrawLine : MonoBehaviour {

    public GameObject target;
    private LineRenderer line;

    void Start()
    {
        line = gameObject.GetComponent< LineRenderer >();
    }

    void Update()
    {
        if (target != null)
        {
            Vector3 p1 = new Vector3(gameObject.transform.position.x, gameObject.transform.position.y, 100f);
            Vector3 p2 = new Vector3(target.transform.position.x, target.transform.position.y, 100f);
            line.SetPosition(0, p1);
            line.SetPosition(1, p2);
            float distance = Vector2.Distance(transform.position, target.transform.position);
            line.material.mainTextureScale = new Vector2(distance, 1);
            line.sortingOrder = 0;
        }
    }
}
